### DISCUSSION TITLE
RFE : Too many decimal places are displayed in the speeds calculated for the segments of a route obtained using WR

### DISCUSSION THREAD

**Discussion by Gilletarom** [rgleason/weather_routing_pi#215] [General]
Hello,

See the screenshot below where we see the properties window of a route built by WR:

![2023-09-24 15_05_28-News](https://github.com/rgleason/weather_routing_pi/assets/6445543/85bd3338-40f1-4663-a989-4eefb8f1b815)

In the column of speeds reached on each segment, we observe that the speeds are written with 5 decimal places. For example, on segment number 1 of the route, the speed, in knots, is 6.92144. (i.e. 12,818.50688 meters/hour, or a precision of the order of a hundredth of a millimeter per hour)
But the speed is in knots. SO, the precision reaches a hundredth of a millimeter per hour.... Taking into account the inaccuracies in the data on winds and currents, and the imprecision of the formula for calculating speeds in WR, this is totally aberrant.

I think that showing a speed of 6.92 knots would even still be too precise because that would be 12,815.84 meters/hour.

Looking more closely, displaying the value 7 knots would provide an accuracy of around one meter per hour (7 knots = 12964 meters per hour). The error introduced would be around 1.1%.

I propose :
1° To please users who will be happy to read speeds to the nearest tenth of a knot, let's keep rounding with one decimal place.
2) Let's keep all the decimal places internally so that in the event of a change in the unit used, this does not add any additional error.

**Comment by Gilletarom** [comment]
And in the same way, in the "ETA" column of a segment, is it reasonable to write durations to the nearest second? For example, referring to the screenshot above, the estimated navigation time on the first segment is 9mn 59s. It would be more honest, and more sensible, to write 10 minutes.

And likewise, the calculated time of passage at the end of this same segment is 12mn 09 s. We could round up to 12:10 p.m. Because making a mistake of at most 10 seconds, on a 12-minutes journey, is that scandalous?

**Comment by sebastien-rosset** [comment]
This issues does not belong to the weather routing plugin. The dialog box you have pasted belongs to OpenCPN core. This issue should be closed, and you can propose a change in OpenCPN core.

**Comment by rgleason** [comment]
Gilletaron. This should be closed and posted in opencpn. Since you are OP can you do that for us.

Thanks, for your observations, testing and suggestions. 
Rick

**Comment by rgleason** [comment]
@Gilletarom Gilletarom??   Will you please close this?  You are the OP!

You should copy this post to OPENCPN ISSUEs.
Closing

**Comment by sebastien-rosset** [comment]
Reopening until we get consensus:

From @Gilletarom :

> Hello Rick Gleason,
> I was preparing to reply to issue https://github.com/rgleason/weather_routing_pi/issues/147, but you closed it. It turns out I had started reading the long series of "buts" you sent me yesterday, particularly the email regarding issue https://github.com/rgleason/weather_routing_pi/issues/147.
>I didn't have time to reply. I should point out that my response has changed between yesterday and today.
> Yesterday, I agreed with Sébastien Rosset, who said that this didn't concern WR but rather the core of OpenCPN.
Today, I no longer agree with Sébastien Rosset.
If we record the waypoints of a route, or those of the route's track, during navigation, we do indeed know the time of passage to the nearest second. On the other hand, when WR creates the "hypothetical track" of a weather route resulting from a search, we don't know exactly the probable time of passage at a point. This is why I find it absurd to display times to the nearest second.
> This is why I requested that the times of passage at points be displayed with only hours and minutes in WR.
> But I hadn't understood that in WR's operation, we were using a display created in the OpenCPN core.
> So, if Sébastien Rosset will forgive me, I say that Sébastien Rosset is not right in what he says. It's in WR's operation that he was going to act. And consequently, when exporting a track from a weather search, and the route resulting from this track, I should have requested that the data be rounded to the nearest hour and minute.

**Comment by sebastien-rosset** [comment]
> If we record the waypoints of a route, or those of the route's track, during navigation, we do indeed know the time of passage to the nearest second. On the other hand, when WR creates the "hypothetical track" of a weather route resulting from a search, we don't know exactly the probable time of passage at a point. T

For this discussion, let's use the OpenCPN terminology:
A **track** is a record of where the vessel has actually been. It's automatically recorded as you sail (typically using GNSS). Tracks show the historical path and are displayed as a colored line on the chart.
A **route** is a planned path you create before sailing. A route can be created by manually placing waypoints to define where you intend to go, or by importing waypoints, including from weather routing. Routes are used for navigation planning and can be followed using the autopilot or manual steering.

When WR exports a routing to OpenCPN core, it's a route. Once a route has been exported to OpenCPN core, you can close and even remove the WR plugin. OpenCPN core manages all aspects of the route, including: visibility; waypoints addition/removal/modification/order changes; overlay and table display of the route; menus; number of digits that are displayed after the decimal.


> I think that showing a speed of 6.92 knots would even still be too precise because that would be 12,815.84 meters/hour.

There seems to be a misunderstanding about precision in different units. When we express a speed as 6.92 knots, converting it to meters per hour (12,815.84 m/h) doesn't magically add precision to our measurement. The conversion creates a number with more digits, but those extra digits don't represent greater precision in the original measurement.

For example, 6 knots converts to exactly 11,112 meters per hour. This longer number doesn't mean we suddenly have a more precise measurement - it's just a mathematical conversion that produces more digits.
Similarly, even if WR reports 6 knots but the display configuration is m/s in OpenCPN, then the route manager will display 3.08667 m/s. There is absolutely nothing the WR plugin can do there. That's the point I was trying to make. Controlling the displayed precision makes sense, but it has to be done in OpenCPN core. We could also have a hybrid approach where numbers are truncated when exporting routes from OpenCPN, and still that won't solve the displayed precision in OpenCPN core.

So for example, even if the WR plugin report 6 knots, the user can still choose to display the speed in a different unit in the route manager. There is nothing the WR plugin can do about the **display precision**.

1. Measurement precision: This refers to the actual precision of the instrument taking the reading, or the WR plugin reporting simulated values.
2. Display precision: This is how we choose to present the measurement to users. Just because we can measure something precisely doesn't mean we should display all those digits.

<img width="628" alt="Image" src="https://github.com/user-attachments/assets/c61645fb-7479-48b8-8114-db85e3802403" />

**Comment by rgleason** [comment]
Guys, This is not really supposed to be a discussion place, but it is useful. I am trying to get this 60 Issue list down to something more manageable and realistic, including checking all items listed as bugs.So can we close this and move it into a Discussion?On Mar 20, 2025 02:55, Sebastien Rosset ***@***.***> wrote:
  Reopening until we get consensus:
From @Gilletarom :

Hello Rick Gleason,
I was preparing to reply to issue #147, but you closed it. It turns out I had started reading the long series of "buts" you sent me yesterday, particularly the email regarding issue #147.
I didn't have time to reply. I should point out that my response has changed between yesterday and today.
Yesterday, I agreed with Sébastien Rosset, who said that this didn't concern WR but rather the core of OpenCPN.
Today, I no longer agree with Sébastien Rosset.
If we record the waypoints of a route, or those of the route's track, during navigation, we do indeed know the time of passage to the nearest second. On the other hand, when WR creates the "hypothetical track" of a weather route resulting from a search, we don't know exactly the probable time of passage at a point. This is why I find it absurd to display times to the nearest second.
This is why I requested that the times of passage at points be displayed with only hours and minutes in WR.
But I hadn't understood that in WR's operation, we were using a display created in the OpenCPN core.
So, if Sébastien Rosset will forgive me, I say that Sébastien Rosset is not right in what he says. It's in WR's operation that he was going to act. And consequently, when exporting a track from a weather search, and the route resulting from this track, I should have requested that the data be rounded to the nearest hour and minute.
—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you modified the open/close state.Message ID: ***@***.***>



sebastien-rosset left a comment (rgleason/weather_routing_pi#147)
Reopening until we get consensus:
From @Gilletarom :

Hello Rick Gleason,
I was preparing to reply to issue #147, but you closed it. It turns out I had started reading the long series of "buts" you sent me yesterday, particularly the email regarding issue #147.
I didn't have time to reply. I should point out that my response has changed between yesterday and today.
Yesterday, I agreed with Sébastien Rosset, who said that this didn't concern WR but rather the core of OpenCPN.
Today, I no longer agree with Sébastien Rosset.
If we record the waypoints of a route, or those of the route's track, during navigation, we do indeed know the time of passage to the nearest second. On the other hand, when WR creates the "hypothetical track" of a weather route resulting from a search, we don't know exactly the probable time of passage at a point. This is why I find it absurd to display times to the nearest second.
This is why I requested that the times of passage at points be displayed with only hours and minutes in WR.
But I hadn't understood that in WR's operation, we were using a display created in the OpenCPN core.
So, if Sébastien Rosset will forgive me, I say that Sébastien Rosset is not right in what he says. It's in WR's operation that he was going to act. And consequently, when exporting a track from a weather search, and the route resulting from this track, I should have requested that the data be rounded to the nearest hour and minute.

**Comment by Gilletarom** [comment]
I'll add to my thoughts.
1. Since the data display tool is located at the heart of OpenCPN, the data transmitted for display must be adapted in WR.
2. For very short or longer route weather forecasts, the accuracy of the schedules should take into account the total distance between the starting point and the arrival point. Thus, for a Channel crossing between England and France, it seems sensible to me not to display the seconds, but for an Atlantic crossing, it seems sensible to give the schedules with even finer precision. For example, we could be satisfied with schedules in 10-minute increments, or why not, hour by hour. The duration of the isochrones requested by the user can also be taken into account. A route built with a one-minute isochrone will be more accurate than one built with a 60-minute isochrone. 
3. I realize that this last reflection is becoming demanding and that choices should be made and explained to WR users if we want to be understood by them.
